File: 844 - Backspace String Compare.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        return getDele(s).equals(getDele(t));
    }
    private String getDele(String s) {
        StringBuilder ret = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (c == '#' && ret.length() > 0) {
                ret.deleteCharAt(ret.length() - 1);
            } else if (c != '#') {
                ret.append(c);
            }
        }
        return ret.toString();
    }
}

File: 844 - Backspace String Compare.cpp
#include <vector>
#include <string>
using namespace std;
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        return getDele(s) == getDele(t);
    }
    string getDele(const string & s) {
        string ret = "";
        for (char c : s) {
            if (c != '#') {
                ret += c;
            } else {
                if (ret.size() > 0) {
                    ret.erase(ret.end() - 1);
                }
            }
        }
        return ret;
    }
};